magenta/protobuf/score.proto

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+syntax = "proto3";
+
+package tensorflow.magenta;
+
+
+// A message for describing a music sequence in terms that are similar to sheet
+// music.
+// It is expected that this message will be populated from a NoteSequence proto
+// that was derived from a score-based sequence (e.g., ABC or MusicXML).
+//
+// This representation is loosely inspired by the MuseScore internal
+// representation format.
+// Tick values are from https://musescore.org/plugin-development/tick-length-values
+// A whole note is 1920 ticks and a 1/64 note is 30 ticks (1920/30).
+message Score {
+  // Vertically-aligned staff elements.
+  repeated Segment segments = 1;
+  // Additional information about the tracks in the score.
+  repeated TrackInfo track_infos = 2;
+}
+
+// Contains additional information about a track.
+// As in MuseScore, a staff has 4 voices and there maybe multiple staves.
+// Therefore: voice = track % 4, staff = track / 4
+message TrackInfo {
+  // The track index.
+  int32 track = 1;
+  // A text description of the track. Examples: "Soprano" or "Trumpet".
+  string name = 2;
+}
+
+// tick values from: 
+
+// in preprocessing, turn ties across measures into longer single notes because
+// we don't want to represent ties between notes in the model.
+
+// A segment represents vertically-aligned staff elements.
+message Segment {
+  // The measure-relative tick position of the staff element. For example if a
+  // chord is at the very beginning of the measure, this would be 0. If it's on
+  // the second beat of a 4/4 measure, it would be 960.
+  int32 measure_local_tick = 1;
```

Comment:
or perhaps "onset_local_tick"? this makes it super clear it's start time and also since we're measuring by ticks, beats are derived and not directly encoded.